### PR TITLE
getRoomActivites

### PR DESCRIPTION
**Pull request by taymoork2**
Implement `getRoomActivities`

[SPARK-98332](https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-98332)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,6 +10,7 @@
     "release:dry-run": "semantic-release --dry-run",
     "start": "nodemon -r dotenv/config scripts/start.js",
     "serve": "parcel scripts/index.html",
+    "serve:start": "parcel scripts/start.html",
```
Comment (lalli-flores) [inline package.json:13]: Not sure we want to add this as part of these PR. Additionally, I don't think we should have 2 "samples" running. Let's discuss this separately

File: package.json [hunk 2]
```diff
@@ -2,7 +2,6 @@
   "name": "@webex/sdk-component-adapter",
   "version": "1.97.2",
   "main": "dist/webexSDKComponentAdapter.esm.js",
-  "module": "dist/webexSDKComponentAdapter.esm.js",
```
Comment (lalli-flores) [inline package.json:5]: Don't remove this property from `package.json`. This is meant to be picked up by tooling that supports module systems in the consumer environment of this library.

File: src/RoomsSDKAdapter.js [hunk 3]
```diff
@@ -2,6 +2,9 @@ import {
   concat,
   from,
   fromEvent,
+  BehaviorSubject,
+  Observable,
+  // throwError,
```
Comment (lalli-flores) [inline src/RoomsSDKAdapter.js:7]: Don't leave commented out code

File: src/RoomsSDKAdapter.js [hunk 4]
```diff
@@ -104,7 +110,10 @@ export default class RoomsSDKAdapter extends RoomsAdapter {
       const room$ = from(this.fetchRoom(ID));
 
       // subscribes to room update events emitted via websocket and emits the updated room object.
-      const roomUpdate$ = fromEvent(this.datasource.rooms, ROOM_UPDATED_EVENT).pipe(
+      const roomUpdate$ = fromEvent(
+        this.datasource.rooms,
+        ROOM_UPDATED_EVENT,
+      ).pipe(
```
Comment (lalli-flores) [inline src/RoomsSDKAdapter.js:116]: Don't add unrelated changes to the contents of the commit

File: src/RoomsSDKAdapter.js [hunk 5]
```diff
@@ -126,12 +135,56 @@ export default class RoomsSDKAdapter extends RoomsAdapter {
       );
 
       // Convert to a multicast observable
-      this.getRoomObservables[ID] = getRoom$.pipe(
-        publishReplay(1),
-        refCount(),
-      );
+      this.getRoomObservables[ID] = getRoom$.pipe(publishReplay(1), refCount());
```
Comment (lalli-flores) [inline src/RoomsSDKAdapter.js:138]: Don't add unrelated changes to the contents of the commit

File: src/RoomsSDKAdapter.js [hunk 6]
```diff
@@ -126,12 +135,56 @@ export default class RoomsSDKAdapter extends RoomsAdapter {
...
     }
 
     return this.getRoomObservables[ID];
   }
+
+  /**
+   * Returns an observable that emits current and future activities from the specified room.
+   *
+   * @param {string} id ID of room to get
```
Comment (lalli-flores) [inline src/RoomsSDKAdapter.js:147]: Keep consistency with the rest of the repo
```suggestion
   * @param {string} ID ID of room to get
```

File: src/RoomsSDKAdapter.js [hunk 7]
```diff
@@ -126,12 +135,56 @@ export default class RoomsSDKAdapter extends RoomsAdapter {
...
+   * @returns {Observable.<Activity>} Observable stream that emits current and future activities from the specified room
+   */
+  getRoomActivities(id) {
```
Comment (lalli-flores) [inline src/RoomsSDKAdapter.js:150]: Keep consistency with the rest of the repo
```suggestion
  getRoomActivities(ID) {
```

File: src/RoomsSDKAdapter.js [hunk 8]
```diff
@@ -126,12 +135,56 @@ export default class RoomsSDKAdapter extends RoomsAdapter {
...
+    logger.debug('ROOM', id, 'getRoomActivities()', ['called with', {id}]);
+
+    // if (!(ID in this.getRoomObservables)) {
```
Comment (lalli-flores) [inline src/RoomsSDKAdapter.js:153]: Remove commented out code

File: src/RoomsSDKAdapter.js [hunk 9]
```diff
@@ -126,12 +135,56 @@ export default class RoomsSDKAdapter extends RoomsAdapter {
...
+    //   return throwError(new Error(`Room ${ID} does not exist`));
+    // }
+
+    const activity$ = new BehaviorSubject(null);
```
Comment (lalli-flores) [inline src/RoomsSDKAdapter.js:157]: Send an empty activity object, not `null`

File: src/RoomsSDKAdapter.js [hunk 10]
```diff
@@ -126,12 +135,56 @@ export default class RoomsSDKAdapter extends RoomsAdapter {
...
+
+    this.datasource.internal.mercury.on(
+      'event:conversation.activity',
+      (activity) => {
+        const {id: hydraId} = deconstructHydraId(id);
```
Comment (lalli-flores) [inline src/RoomsSDKAdapter.js:162]: Well, this is not really a hydra ID anymore, this is the UUID at this point.
```suggestion
        const {id: UUID} = deconstructHydraId(id);
```

File: src/RoomsSDKAdapter.js [hunk 11]
```diff
@@ -126,12 +135,56 @@ export default class RoomsSDKAdapter extends RoomsAdapter {
...
+
+        if (activity.target && activity.target.id === hydraId) {
+          const activityInterface = {
+            id: activity.id,
+            roomId: activity.target.id,
+            verb: activity.verb,
+            content: activity.object,
+            contentType: activity.object.objectType,
+            personId: activity.actor.id,
+            displayAuthor: !!activity.object.displayName,
```
Comment (lalli-flores) [inline src/RoomsSDKAdapter.js:172]: Set this to `false` always. UI can decide when to show

File: src/RoomsSDKAdapter.js [hunk 12]
```diff
@@ -126,12 +135,56 @@ export default class RoomsSDKAdapter extends RoomsAdapter {
...
+      'event:conversation.activity',
+      (activity) => {
+        const {id: hydraId} = deconstructHydraId(id);
+
+        if (activity.target && activity.target.id === hydraId) {
+          const activityInterface = {
+            id: activity.id,
+            roomId: activity.target.id,
+            verb: activity.verb,
+            content: activity.object,
```
Comment (lalli-flores) [inline src/RoomsSDKAdapter.js:169]: Update the `Activity` object structure as well in the [interfaces repo](https://github.com/webex/component-adapter-interfaces)

File: src/RoomsSDKAdapter.js [hunk 13]
```diff
@@ -126,12 +135,56 @@ export default class RoomsSDKAdapter extends RoomsAdapter {
...
+
+    this.datasource.internal.mercury.on(
+      'event:conversation.activity',
+      (activity) => {
+        const {id: hydraId} = deconstructHydraId(id);
+
+        if (activity.target && activity.target.id === hydraId) {
+          const activityInterface = {
+            id: activity.id,
+            roomId: activity.target.id,
```
Comment (lalli-flores) [inline src/RoomsSDKAdapter.js:167]: This is not the expected attribute
```suggestion
            roomID: activity.target.id,
```

File: src/RoomsSDKAdapter.js [hunk 14]
```diff
@@ -126,12 +135,56 @@ export default class RoomsSDKAdapter extends RoomsAdapter {
...
+    const activity$ = new BehaviorSubject(null);
+
+    this.datasource.internal.mercury.on(
+      'event:conversation.activity',
+      (activity) => {
+        const {id: hydraId} = deconstructHydraId(id);
+
+        if (activity.target && activity.target.id === hydraId) {
+          const activityInterface = {
+            id: activity.id,
```
Comment (lalli-flores) [inline src/RoomsSDKAdapter.js:166]: This is not the expected attribute
```suggestion
            ID: activity.id,
```

File: src/RoomsSDKAdapter.js [hunk 15]
```diff
@@ -126,12 +135,56 @@ export default class RoomsSDKAdapter extends RoomsAdapter {
...
+            roomId: activity.target.id,
+            verb: activity.verb,
+            content: activity.object,
+            contentType: activity.object.objectType,
+            personId: activity.actor.id,
```
Comment (lalli-flores) [inline src/RoomsSDKAdapter.js:171]: This is not the expected attribute
```suggestion
            personID: activity.actor.id,
```

File: src/RoomsSDKAdapter.js [hunk 16]
```diff
@@ -126,12 +135,56 @@ export default class RoomsSDKAdapter extends RoomsAdapter {
...
+
+    const activity$ = new BehaviorSubject(null);
+
+    this.datasource.internal.mercury.on(
+      'event:conversation.activity',
+      (activity) => {
+        const {id: hydraId} = deconstructHydraId(id);
+
+        if (activity.target && activity.target.id === hydraId) {
+          const activityInterface = {
```
Comment (lalli-flores) [inline src/RoomsSDKAdapter.js:165]: This is not really an interface at this point, so this variable name is misleading. This is more like an activity of the component system
```suggestion
          const componentActivity = {
```

Comment (lalli-flores) [reply src/RoomsSDKAdapter.js:165]: Alternatively, you can just simply call this "activity" and the one from the SDK "sdkActivity". This is the pattern we follow in other modules
```suggestion
          const activity = {
```

File: src/RoomsSDKAdapter.js [hunk 17]
```diff
@@ -126,12 +135,56 @@ export default class RoomsSDKAdapter extends RoomsAdapter {
...
+            created: activity.published,
+          };
+
+          activity$.next(activityInterface);
+        }
+      },
+    );
+
+    return activity$.pipe(
+      // skip(1), // causes test case to fail, use filter instead to skip first value
```
Comment (lalli-flores) [inline src/RoomsSDKAdapter.js:182]: Remove commented code

File: src/RoomsSDKAdapter.js [hunk 18]
```diff
@@ -126,12 +135,56 @@ export default class RoomsSDKAdapter extends RoomsAdapter {
...
+      filter((activity) => activity !== null),
```
Comment (lalli-flores) [inline src/RoomsSDKAdapter.js:183]: Not sure i understand why you're filtering

**Comment by lalli-flores** [comment]
Also, in terms of commits, tests are not separated from the code in introduces, it's all once change. Also, please use a complete sentence as a commit message.

File: src/RoomsSDKAdapter.js (hunk 8 above)
Comment (taymoork2) [reply src/RoomsSDKAdapter.js:153]: Do we want this to follow the cache pattern also being done in `getRooms`?

File: src/RoomsSDKAdapter.js (hunk 18 above)
Comment (taymoork2) [reply src/RoomsSDKAdapter.js:183]: Filtering is done so it doesnt send an empty object when you first subscribe

Comment (lalli-flores) [reply src/RoomsSDKAdapter.js]: it shouldn't send `null` ever, it should always send an activity object

**Review by lalli-flores — APPROVED** [review]
